File: Backend/PortfolioAPI/Models/Entities/Project.cs
namespace PortfolioAPI.Models;

public class Project
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? TitleAr { get; set; }
    public string? Description { get; set; }
    public string? DescriptionAr { get; set; }
    public string Type { get; set; } = "Design";
    public string? Category { get; set; }
    public string? TechStack { get; set; }
    public string? LiveUrl { get; set; }
    public string? Problem { get; set; }
    public string? ProblemAr { get; set; }
    public string? Solution { get; set; }
    public string? SolutionAr { get; set; }
    public string? Role { get; set; }
    public string? RoleAr { get; set; }
    public string? Impact { get; set; }
    public string? ImpactAr { get; set; }
    public int SortOrder { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public List<Media> Media { get; set; } = new();
}

File: src/core/types/project.ts
export interface MediaItem {
  id: number;
  mediaType: 'Image' | '3DModel';
  url: string;
  fileName?: string;
  isPrimary: boolean;
}

export interface Project {
  id: number;
  title: string;
  titleAr?: string;
  description?: string;
  descriptionAr?: string;
  type: 'Design' | 'Code' | 'Full-stack';
  category?: string;
  techStack?: string;
  liveUrl?: string;
  sortOrder: number;
  media: MediaItem[];
  problem?: string;
  problemAr?: string;
  solution?: string;
  solutionAr?: string;
  role?: string;
  roleAr?: string;
  impact?: string;
  impactAr?: string;
}
